src/Pages/3DPrintingConsole/3DConsole.js

```diff
@@ -68,13 +57,13 @@ export default class PrintConsole3D extends React.Component {
   Search for object in db using its name and date
   Set new progress = event value
   */
-  async updateProgress (jsonObject, event) {
+  handleUpdateProgress = async (requestToUpdate, event) => {
     const newProgress = event.target.value
-    if (newProgress === jsonObject.progress) return
+    if (newProgress === requestToUpdate.progress) return
     const updateRequest = {
       progress: newProgress,
-      email: jsonObject.email,
-      date: jsonObject.date
+      email: requestToUpdate.email,
+      date: requestToUpdate.date
     }
     await update3DPrintRequestProgress(updateRequest, this.state.authToken)
     this.callDatabase() // reload database
```

Comment:
I know you didn't write this, but let's also avoid `callDatabase` when we dont need it. We can filter and update an array based on the `_id` field like [this stackoverflow answer](https://stackoverflow.com/a/35206193)